src/databricks/labs/lsql/backends.py

```diff
@@ -177,23 +156,45 @@ def _row_to_sql(cls, row: DataclassInstance, fields: tuple[dataclasses.Field[Any
         data = []
         for f in fields:
             value = getattr(row, f.name)
-            field_type = cls._field_type(f)
-            if isinstance(field_type, UnionType):
-                field_type = field_type.__args__[0]
-            if value is None:
-                data.append("NULL")
-            elif field_type is bool:
-                data.append("TRUE" if value else "FALSE")
-            elif field_type is str:
-                value = str(value).replace("'", "''")
-                data.append(f"'{value}'")
-            elif field_type is int:
-                data.append(f"{value}")
-            else:
-                msg = f"unknown type: {field_type}"
-                raise ValueError(msg)
+            data.append(cls._value_to_sql(value))
         return ", ".join(data)
 
+    @classmethod
+    def _value_to_sql(cls, value: Any) -> str:
```

Comment:
I understand this approach, and that it's not being introduced in this PR but is rather the existing way things work. However I would like to note that this approach does leave itself open to type confusion because the schema (and types) depends on the typing hints whereas the control flow for conversion depends on the values and we never verify that the values have the expected types.

Is this harmless in this context? (Can it be abused? I'm not sure. Because the context is kind of open, this is hard to verify and that often makes people nervous. But this is why most modern serde frameworks use the schema as the blueprint during serialisation and deserialisation rather than the relying on or trusting the runtime types of values.)

Reply:
PR is welcome ;)